test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -622,14 +622,21 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 		})
 
 		It("should have core kube-system componentry running", func() {
-			coreComponents := []string{"kube-proxy", "kube-addon-manager", common.APIServerComponentName, "kube-controller-manager", "kube-scheduler"}
+			coreComponents := []string{
+				common.AddonManagerComponentName,
+				common.APIServerComponentName,
+				common.ControllerManagerComponentName,
+				common.KubeProxyAddonName,
+				common.SchedulerComponentName,
+			}
 			if to.Bool(eng.ExpandedDefinition.Properties.OrchestratorProfile.KubernetesConfig.UseCloudControllerManager) {
-				coreComponents = append(coreComponents, "cloud-controller-manager")
-				if common.IsKubernetesVersionGe(eng.ExpandedDefinition.Properties.OrchestratorProfile.OrchestratorVersion, "1.13.0") {
+				coreComponents = append(coreComponents, common.CloudControllerManagerComponentName)
+				if eng.ExpandedDefinition.Properties.ShouldEnableAzureCloudAddon(common.AzureDiskCSIDriverAddonName) &&
```

Comment:
Should we re-use the `ShouldEnableAzureCloudAddon` method in E2E? Is there a reason why we can't simply refer to the state of the api model to see whether or not the addon is enabled or not?

Reply:
I don't think we will need to reuse it since those addons are turned on/off based on `ShouldEnableAzureCloudAddon` anyways. I will change it to `eng.HasAddon()`.